gnat2goto/driver/tree_walk.adb

```diff
@@ -4543,21 +4543,54 @@ package body Tree_Walk is
          Target_Type_Irep : constant Irep :=
            Follow_Symbol_Type (Get_Type (Target_Name), Global_Symbol_Table);
       begin
-         if Kind (Target_Type_Irep) in Class_Type then
-            if Attr_Id = "size" then
-
-               --  Just check that the front-end already applied this size
-               --  clause, i .e. that the size of type-irep we already had
-               --  equals the entity type this clause is applied to (and the
-               --  size specified in this clause).
-               pragma Assert (Entity_Esize = Get_Width (Target_Type_Irep)
-                              and Entity_Esize =
-                                Integer (UI_To_Int (Intval (Expression (N)))));
-               return;
-            end if;
-         end if;
-         Report_Unhandled_Node_Empty (N, "Process_Declaration",
+         pragma Assert (Kind (Target_Type_Irep) in Class_Type);
+         if Attr_Id = "size" then
+            --  Just check that the front-end already applied this size
+            --  clause, i .e. that the size of type-irep we already had
+            --  equals the entity type this clause is applied to (and the
+            --  size specified in this clause).
+            pragma Assert (Entity_Esize = Get_Width (Target_Type_Irep)
+                           and Entity_Esize =
+                             Integer (UI_To_Int (Intval (Expression (N)))));
+         elsif Attr_Id = "address" then
+            --  Assuming this Ada code:
+            --------------------
+            --  Var : VarType;
+            --  for Var'Address use System'To_Address (hex_address);
+            --------------------
+            --
+            --  Produce this C code:
+            --------------------
+            --  &Var = (VarType*)system__to_address(hex_address);
```

Comment:
Is the `system__to_address` here necessary? Seems to me that just casting the hex address directly to a pointer should work just as well.

I also don't really know what this means, addresses aren't assignable in C proper. Are addresses assignable in GOTO code?

Reply:
You mean like this?:
```
#include <assert.h>

int main() {
  int *var = (int*)0x40001000;

  *var = 4;
  assert(*(int*)0x40001000==4);
  assert(var==0x40001000);
  assert(*var==4);

  return 0;
}

```